onnxruntime/core/providers/cpu/object_detection/non_max_suppression_helper.h

```diff
@@ -30,7 +30,7 @@ struct PrepareContext {
   const float* iou_threshold_ = nullptr;
   int64_t num_batches_ = 0;
   int64_t num_classes_ = 0;
-  int64_t num_boxes_ = 0;
+  int num_boxes_ = 0;
```

Comment:
>int [](start = 2, length = 3)

Int is a bad data type for us

Reply:
The dim3 type in CUDA only support int32. 

```c
struct __device_builtin__ dim3
{
dim3(unsigned int vx = 1, unsigned int vy = 1, unsigned int vz = 1) : x(vx), y(vy), z(vz) {}
};
```

The dim3 type or int  are used for calculating the number of threads per block and the number of blocks per grid when invoking a kernel. So this is a very general, and we can't say using "int" is bad.